### DISCUSSION TITLE
Render config Netbox v3.6.4

### DISCUSSION THREAD

**Discussion by RandyRozo** [netbox-community/netbox#16172] [Help Wanted!]
**Netbox v3.6.4** 

I have the following jinja2 template but it is not rendering, I do not get the hostname and I get the following error in 'interfaces'

**Config Template**

hostname {{ device.name }}

{% for interface in device.interfaces.all() %}
interface {{ interface.name }} 
 switchport mode access 
 switchport access vlan {{ interface.untagged_vlan.vid }} 
{% endfor %}

**Error**

Traceback (most recent call last):
  File "/opt/netbox/netbox/dcim/views.py", line 2071, in get_extra_context
    rendered_config = config_template.render(context=context_data)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/netbox/netbox/extras/models/configs.py", line 280, in render
    output = template.render(**_context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/netbox/venv/lib/python3.11/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/opt/netbox/venv/lib/python3.11/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "<template>", line 3, in top-level template code
  File "/opt/netbox/venv/lib/python3.11/site-packages/jinja2/sandbox.py", line 326, in getattr
    value = getattr(obj, attribute)
            ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'str object' has no attribute 'interfaces'

**Comment by candlerb** [comment]
That template works for me in Netbox 3.7.8 / Python 3.8.10.

Is that the *entire* template, or is there some other part before this? Somehow "device" seems to be set to a string, not an object.

Can you replicate this at demo.netbox.dev ?

**Reply by RandyRozo** [reply]
That's all the template I'm using, I replicated it in demo.netbox.dev and it works for me.

Do you know by chance why 'device' is configured as a string and not an object, is it a version problem?

**Reply by candlerb** [reply]
I don't know, but you should in any case upgrade to v3.6.9 to eliminate that possibility.

Have you got any plugins installed? Badly-behaved ones might mess with Netbox internals. Try disabling them temporarily.
